ocean_lib/ocean/ocean_assets.py

```diff
@@ -140,25 +137,96 @@ def find_service_by_data_token(self, data_token: str, services: list) -> str:
     def build_data_tokens_list(
         self, services: list, deployed_erc20_tokens: list
     ) -> list:
-        data_tokens = []
+        datatokens = []
+        # (1-n) service per datatoken, 1 datatoken per service
         for erc20_token in deployed_erc20_tokens:
-            for service in services:
-                data_token = {
+            datatokens = datatokens + [
+                {
                     "address": erc20_token.address,
                     "name": erc20_token.contract.caller.name(),
                     "symbol": erc20_token.symbol(),
                     "serviceId": service.id,
                 }
-                data_tokens.append(data_token)
-        return data_tokens
+                for service in services
+                if service.data_token == erc20_token.address
+            ]
+
+        return datatokens
+
+    @staticmethod
+    def _encrypt_ddo_locally(
```

Comment:
Please consider renaming to `_encrypt_ddo`.  The `_locally` seems like a misnomer given that you're still making a call to the provider encrypt endpoint.